### PR TITLE
Log repo_info.json File in entrypoint.sh

### PR DESCRIPTION
**Pull request by rohankh532**
For issue #196

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -47,7 +47,13 @@ export ENABLED_CHECKS=
 #
 # Boolean inputs are strings https://github.com/actions/runner/issues/1483.
 # ===============================================================================
-curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/$GITHUB_REPOSITORY > repo_info.json
+status_code=$(curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/"$GITHUB_REPOSITORY" -o repo_info.json -w '%{http_code}')
+if [[ $status_code -ge 300 ]]; then
```
Comment (laurentsimon) [inline entrypoint.sh:51]: also check for <200

File: entrypoint.sh [hunk 2]
```diff
@@ -47,7 +47,13 @@ export ENABLED_CHECKS=
...
+    error_msg=$(jq -r .message repo_info.json 2>/dev/null || echo 'unknown error')
+    echo "Failed to get repository information from GitHub, response $status_code: $error_msg"
+    exit 1;
+fi
+
 export SCORECARD_PRIVATE_REPOSITORY="$(cat repo_info.json | jq -r '.private')"
 export SCORECARD_DEFAULT_BRANCH="refs/heads/$(cat repo_info.json | jq -r '.default_branch')"
 export SCORECARD_IS_FORK="$(cat repo_info.json | jq -r '.fork')"
```
Comment (laurentsimon) [inline entrypoint.sh:59]: we want to print the entire repo_info.json to help troubleshooting, then we can remove the file

File: entrypoint.sh (hunk 1 above)
Comment (laurentsimon) [reply entrypoint.sh:51]: example https://github.com/slsa-framework/slsa-github-generator/pull/38/files

File: entrypoint.sh (hunk 2 above)
Comment (rohankh532) [reply entrypoint.sh:60]: fixed

File: entrypoint.sh [hunk 3]
```diff
@@ -47,7 +47,14 @@ export ENABLED_CHECKS=
 #
 # Boolean inputs are strings https://github.com/actions/runner/issues/1483.
 # ===============================================================================
-curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/$GITHUB_REPOSITORY > repo_info.json
+status_code=$(curl -s -H "Authorization: Bearer $GITHUB_AUTH_TOKEN" https://api.github.com/repos/"$GITHUB_REPOSITORY" -o repo_info.json -w '%{http_code}')
+if [[ $status_code -lt 200 ]] || [[ $status_code -ge 300 ]]; then
+    error_msg=$(jq -r .message repo_info.json 2>/dev/null || echo 'unknown error')
+    echo "Failed to get repository information from GitHub, response $status_code: $error_msg"
+    echo "$(<repo_info.json)"
```
Comment (laurentsimon) [inline entrypoint.sh:54]: this looks clever :-) `cat "$file"` would have worked too I think, Want to remove the file?

**Review by laurentsimon — APPROVED** [review]

File: entrypoint.sh (hunk 3 above)
Comment (laurentsimon) [inline entrypoint.sh:54]: note: probably fine to display the content all the times, not just on failure. I think what you have works, but I'm always worried about some unexpected corner cases where the data may still be valuable

**Review by laurentsimon — APPROVED** [review]

**Review by laurentsimon — APPROVED** [review]
Thanks

**Pull request merged**